// fogg-utils/dist/is/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.isBrowser = exports.isUndefined = exports.isString = exports.isPasswordMatch = exports.isPassword = exports.isObject = exports.isJson = exports.isFunction = exports.isNumber = exports.isFalse = exports.isDefined = exports.isArray = void 0;

function _typeof(obj) { if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

var isArray = function isArray(variable) {
  return Array.isArray(variable);
};

exports.isArray = isArray;

var isDefined = function isDefined(variable) {
  return typeof variable !== 'undefined' && variable !== null;
};

exports.isDefined = isDefined;

var isFalse = function isFalse(variable) {
  return isDefined(variable) && variable === false;
};

exports.isFalse = isFalse;

var isNumber = function isNumber(variable) {
  return typeof variable === 'number';
};

exports.isNumber = isNumber;

var isFunction = function isFunction(variable) {
  return typeof variable === 'function';
};

exports.isFunction = isFunction;

var isJson = function isJson(str) {
  if (!str || str === null) {
    return false;
  }

  try {
    JSON.parse(str);
  } catch (e) {
    return false;
  }

  return true;
};

exports.isJson = isJson;

var isObject = function isObject(variable) {
  return isDefined(variable) && _typeof(variable) === 'object' && !Array.isArray(variable);
};

exports.isObject = isObject;

var isPassword = function isPassword(password) {
  var min = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : 8;
  return password && password.length >= min;
};

exports.isPassword = isPassword;

var isPasswordMatch = function isPasswordMatch(p1, p2) {
  return isPassword(p1) && isPassword(p2) && p1 === p2;
};

exports.isPasswordMatch = isPasswordMatch;

var isString = function isString(variable) {
  return isDefined(variable) && typeof variable === 'string';
};

exports.isString = isString;

var isUndefined = function isUndefined(variable) {
  return typeof variable === 'undefined' || variable === null;
};

exports.isUndefined = isUndefined;

var isBrowser = function isBrowser() {
  return typeof window !== 'undefined';
};

exports.isBrowser = isBrowser;
