// fogg-utils/dist/graphql/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.validateFields = exports.getGraphQlError = void 0;

var getGraphQlError = function getGraphQlError(error) {
  return {
    error: true,
    message: error.toString().replace('Error: GraphQL error: ', '')
  };
};

exports.getGraphQlError = getGraphQlError;

var validateFields = function validateFields(validations, values) {
  var alert = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : 'Error trying to create the item';
  var messages = {};
  Object.keys(validations).forEach(function (field) {
    var _validations$field = validations[field],
        len = _validations$field.len,
        isEmpty = _validations$field.isEmpty;

    if (len && len.arg) {
      if (values[field].length !== len.arg) {
        messages[field] = {
          msg: len.msg
        };
      }
    }

    if (isEmpty) {
      if (values[field] === '' || !values[field]) {
        messages[field] = {
          msg: isEmpty.msg
        };
      }
    }
  });
  return Object.keys(messages).length > 0 ? {
    error: true,
    alert: alert,
    messages: messages
  } : {
    error: false
  };
};

exports.validateFields = validateFields;
